Tests/IntegrationTests.Shared/RealmResults/SimpleLINQtests.cs

```diff
@@ -361,6 +373,91 @@ public void FirstWorks()
 
 
         [Test]
+        public void FirstOrDefaultReturnsDefault()
+        {
+            var expectedDef = _realm.All<Person>().FirstOrDefault(p => p.FirstName == "Zaphod");
+            Assert.That(expectedDef, Is.Null);
+
+            expectedDef = _realm.All<Person>().Where(p => p.FirstName == "Just Some Guy").FirstOrDefault();
+            Assert.That(expectedDef, Is.Null);
+        }
+
+
+        [Test]
+        public void LastFailsToFind()
+        {
+            Assert.Throws<InvalidOperationException>(() => _realm.All<Person>().Last(p => p.Latitude > 100));
```

Comment:
I don't think we need 4 test cases for that - it will either fail with the first two or it will not :)